src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertyBuilder.java

```diff
@@ -946,6 +946,20 @@ public JsonProperty.Access removeNonVisible(boolean inferMutators,
         if (acc == null) {
             acc = JsonProperty.Access.AUTO;
         }
+
+        // [databind#2951] add feature to inverse access logic
```

Comment:
I think this should actually be moved to occur within "findAccess()" -- I can change that